### PR TITLE
[WIP] Mobile cloud support

### PR DESCRIPTION
**Pull request by jessicafalk**
#### Description
adding mobile cloud support to the FPS project

- [x] Android
- [ ] iOS

#### Tests
running it locally and in cloud for Android (can't test iOS right now :( )

#### Documentation
will follow later

**Did you remember a changelog entry?** yes

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs [hunk 1]
```diff
@@ -23,6 +22,7 @@ public class AndroidWorkerConnector : MobileWorkerConnector, ITileProvider
         private const string Small = "small";
         private const string Large = "large";
 
+        public bool ShouldConnectLocally;
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: considering to move this into the MobileWorkerConnector as it is difficult to determine whether the app should connect locally or to a cloud deployment without relying on command line arguments

File: .gitignore [hunk 2]
```diff
@@ -5,6 +5,7 @@
 .idea/
 .vs/
 .DS_Store
+DevAuthToken.txt*
```
Comment (jamiebrynes7) [inline .gitignore]: Why the wildcard at the end? Is this stored in the root project directory?

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs [hunk 3]
```diff
@@ -37,22 +37,30 @@ public class AndroidWorkerConnector : MobileWorkerConnector, ITileProvider
...
+            if (!ShouldConnectLocally)
+            {
+                var textAsset = Resources.Load<TextAsset>("DevAuthToken");
+                if (textAsset != null)
+                {
+                    DevelopmentAuthToken = textAsset.text.Trim();
+                }
+                else
+                {
+                    Debug.LogWarning("Unable to find DevAuthToken.txt in the Resources folder.");
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: Do we plan to add tooling to generate these easily? Currently a user has to invoke:

```
spatial project auth dev-auth-token create --description "My token" | jq <some_parsing_magic_here> > workers/unity/Assets/Resources/DevAuthToken.txt
```

File: .gitignore (hunk 2 above)
Comment (jessicafalk) [reply .gitignore]: because of the meta file that unity creates

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs (hunk 3 above)
Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: I would like to add some kind of tooling for this. maybe a simple menu button for now that will just generate a token that is valid for a few days.

File: .gitignore (hunk 2 above)
Comment (jamiebrynes7) [reply .gitignore]: :(

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs (hunk 3 above)
Comment (jamiebrynes7) [reply workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: Should be a quick one to build as well?

File: cloud_launch_small.json [hunk 4]
```diff
@@ -48,6 +48,54 @@
...
+          "name": "world_size",
+          "value": "small"
+        }
+      ],
+      "permissions": [
+        {
+          "all": {}
+        }
+      ],
+      "load_balancing": {
```
Comment (jamiebrynes7) [inline cloud_launch_small.json]: These need to be removed from `iOSClient` and `AndroidClient`

File: cloud_launch_large.json [hunk 5]
```diff
@@ -55,6 +55,48 @@
...
+        }
+      ],
+      "permissions": [
+        {
+          "all": {}
+        }
+      ],
+    },
+    {
+      "worker_type": "SimulatedPlayer",
```
Comment (jamiebrynes7) [inline cloud_launch_large.json]: Oh this needs to be removed.

Comment (jamiebrynes7) [reply cloud_launch_large.json]: On both

**Pull request merged**